packages/core/src/metrics/MetricsService.ts

```diff
@@ -1,72 +1,107 @@
 import { ServiceConfiguration } from "../configuration";
-import { Metrics, MetricsPayload, MetricsType } from "./model";
+import { Metrics, MetricsPayload } from "./model";
+import { CordovaAppMetrics } from "./platform/CordovaAppMetrics";
+import { CordovaDeviceMetrics } from "./platform/CordovaDeviceMetrics";
 import { MetricsPublisher, NetworkMetricsPublisher } from "./publisher";
+import { isMobileCordova } from "./platform/PlatformUtils";
+
+declare var window: any;
 
 /**
  * AeroGear Services metrics service
  */
 export abstract class MetricsService {
 
-    private publisher: MetricsPublisher;
+  public static readonly CLIENT_ID_KEY = "aerogear_metrics_client_key";
+  public static readonly DEFAULT_METRICS_TYPE = "init";
 
-    constructor(private readonly configuration: ServiceConfiguration) {
-        this.publisher = new NetworkMetricsPublisher(configuration.url);
-    }
+  private publisher: MetricsPublisher;
 
-    set metricsPublisher(publisher: MetricsPublisher) {
-        this.publisher = publisher;
-    }
+  private readonly defaultMetrics: Metrics[];
 
-    get metricsPublisher(): MetricsPublisher {
-        return this.publisher;
-    }
+  constructor(private readonly configuration: ServiceConfiguration) {
+    this.publisher = new NetworkMetricsPublisher(configuration.url);
+    this.defaultMetrics = this.buildDefaultMetrics();
+  }
+
+  set metricsPublisher(publisher: MetricsPublisher) {
+    this.publisher = publisher;
+  }
+
+  get metricsPublisher(): MetricsPublisher {
+    return this.publisher;
+  }
+
+  /**
+   * Collect metrics for all active metrics collectors
+   * Send data using metrics publisher
+   */
+  // tslint:disable-next-line:member-ordering
+  public sendAppAndDeviceMetrics(): Promise<any> {
+    return this.publish(MetricsService.DEFAULT_METRICS_TYPE, []);
+  }
 
-    /**
-     * Collect metrics for all active metrics collectors
-     * Send data using metrics publisher
-     */
-    public abstract sendAppAndDeviceMetrics(): Promise<any>;
-
-    /**
-     * Publish metrics using predefined publisher
-     *
-     * @param type type of the metrics to be published
-     * @param metrics metrics instances that should be published
-     */
-    public publish(type: MetricsType, metrics: Metrics[]): Promise<any> {
-        if (!type) {
-            throw new Error(`Type is invalid: ${type}`);
-        }
-
-        const payload: MetricsPayload = {
-            clientId: this.getClientId(),
-            type,
-            timestamp: new Date().getTime(),
-            data: {}
-        };
-
-        metrics.forEach(m => {
-            payload.data[m.identifier] = m.collect();
-        });
-
-        return this.publisher.publish(payload);
+  /**
+   * Publish metrics using predefined publisher
+   *
+   * @param type type of the metrics to be published
+   * @param metrics metrics instances that should be published
+   */
+  public publish(type: string, metrics: Metrics[]): Promise<any> {
+    if (!type) {
```

Comment:
can replace this with an [enum](https://www.typescriptlang.org/docs/handbook/enums.html) in the future

Reply:
Actually we had that as interface before. I removed that (see left side).
It was because the metrics side doesn't know what other SDK will use as key and defining that in core will mean that people will need to update core to use and pass new type which is really bad.